Implement Charge and more abilities

TempsRay opened the pull request:
Implemented Charge, modifying the `TypeBoostTag` that was only being used by Flash Fire

Implemented:
* Wind Power
* Electromorphosis
* Thermal Exchange

Partial Implementation:
* Wind Rider (needs tailwind)
* Wind Power (needs tailwind)
* Victory Star (needs to affect allies)
* Sweet Veil (needs to affect allies)
* Rattled (does not trigger on intimidate)

Flashfyre reviewed (changes requested):
Update for i18n changes

TempsRay commented:
Also fixed an issue where Beedrill did not properly learn fury cutter at level 11. I removed the wind tag from Sandstorm since despite being classified as a wind move in-game, it does not interact with wind power and wind rider
